src/Core/Framework/Bundle.php

```diff
@@ -158,4 +162,37 @@ protected function registerContainerFile(ContainerBuilder $container): void
         $loader = new XmlFileLoader($container, new FileLocator($this->getPath()));
         $loader->load($containerFilePath);
     }
+
+    protected function registerLogger(ContainerBuilder $container): void
+    {
+        $servicePrefix = $this->getContainerPrefix();
+        $parameterNameLoggerMaxFiles = sprintf('%s.logger.max_files', $servicePrefix);
+
+        if (!$container->hasParameter($logMaxFiles = $parameterNameLoggerMaxFiles)) {
+            $container->setParameter($logMaxFiles, 14);
+        }
+
+        $serviceIdLoggerHandler = sprintf('%s.logger_handler', $servicePrefix);
+        $serviceIdLoggerFormatter = sprintf('%s.logger_formatter', $servicePrefix);
+        $serviceIdLogger = sprintf('%s.logger', $servicePrefix);
+
+        $loggerFormaterService = (new Definition(PsrLogMessageProcessor::class))->setPublic(false);
+        $loggerHandlerService = (new Definition(
+                RotatingFileHandler::class,
+                [
+                    sprintf('%%kernel.logs_dir%%/%s_%%kernel.environment%%.log', $servicePrefix),
+                    sprintf('%%%s%%', $parameterNameLoggerMaxFiles),
+                ]
+            ))
+            ->setPublic(false)
+        ;
+        $loggerService = (new Definition(Logger::class, [$servicePrefix]))
```

Comment:
The logger itself should be also private. Enforce people to inject all services 😉

Reply:
On the other hand I can not extend an other bundle and also write into its log files. My example is not cool but limiting this service is not cool either in my POV.